File: app/src/main/java/com/example/facedetect/SCRFDNcnn.java

package com.example.facedetect;

import android.content.res.AssetManager;
import android.view.Surface;

public class SCRFDNcnn
{
    public native boolean loadModel(AssetManager mgr, int modelid, int cpugpu);
    public native boolean openCamera(int facing);
    public native boolean closeCamera();
    public native boolean setOutputWindow(Surface surface);

    static {
        System.loadLibrary("scrfdncnn");
    }
}

File: app/src/main/java/com/example/facedetect/SCRFDNcnn.kt
package com.example.facedetect

import android.content.res.AssetManager
import android.view.Surface

class SCRFDNcnn {
    external fun loadModel(mgr: AssetManager?, modelid: Int, cpugpu: Int,dir:String): Boolean

    external fun openCamera(facing: Int): Boolean

    external fun closeCamera(): Boolean

    external fun setOutputWindow(surface: Surface?): Boolean

    companion object {
        init {
            System.loadLibrary("scrfdncnn")
        }
    }
}
